src/rest_api/api/indexing/bulk_index.rs

```diff
@@ -139,13 +138,12 @@ fn bulk_request_doc_to_tantivy_doc(
     );
 
     let scaffolds = &PARSED_SCAFFOLDS;
-    let scaffold_matches = scaffold_search(&tautomer, scaffolds).map_err(|err| err.to_string())?;
+    let scaffold_matches = scaffold_search(&tautomer, scaffolds)?;
 
     let mut scaffold_json = Value::Null;
     if !scaffold_matches.is_empty() {
         scaffold_json =
-            serde_json::from_str(format!(r#"{{ "scaffolds": {:?} }}"#, scaffold_matches).as_str())
-                .map_err(|err| err.to_string())?;
+            serde_json::from_str(format!(r#"{{ "scaffolds": {:?} }}"#, scaffold_matches).as_str())?;
```

Comment:
you can actually skip the string/parse round trip (which is quite a bit of machinery). `serde_json::json!` will build the Value through a macro at compile time.

See more: https://docs.rs/serde_json/latest/serde_json/macro.json.html